ui/src/app/app-routing.module.ts

```diff
@@ -170,16 +170,18 @@ const routes: Routes = [
         // The notebook redirect pages are interstitial pages, so we want to
         // give them special chrome treatment - we therefore put them outside
         // the normal /workspaces hierarchy.
-        path: 'workspaces/:ns/:wsid/notebooks/create',
+        path: 'workspaces/:ns/:wsid/notebooks/create/:nbName/:kernelType',
```

Comment:
These feel like query parameters to me. Both could be optional.

Reply:
I don't think we want them to be optional? I think part of the goal is that you need to pick a name and kernel when you create a new notebook. When would we not need them?

Reply:
Whether we want the name to be optional can be a product question. Whether we want the kernel to be optional, I'm not sure if we'll have other links to create a new notebook, but having a default value as we currently do might make sense in some of those cases.

I still think these don't feel correct as path parameters. Name/kernel type don't seem natural in a REST URL, rather than referring to an existing resource, they are more part of the creation payload (we're referencing a thing that doesn't yet exist). It feels like we could have a potentially growing list of values here (e.g. if we added more options to your modal). Hence why query parameters feel more natural to me.

Reply:
That makes sense. I can convert them to query parameters.